lib/text/cue.js

```diff
@@ -344,15 +344,29 @@ shaka.text.Cue = class {
   clone() {
     const clone = new shaka.text.Cue(0, 0, '');
 
-    for (const k in this) {
-      clone[k] = this[k];
+    /**
+     * Deep clone helper
+     * @param {*} value
+     * @return {*}
+     */
+    const deepClone = (value) => {
+      if (value === null || typeof value !== 'object') {
+        return value;
```

Comment:
Can't we use `shaka.util.ObjectUtils.cloneObject()`?

Reply:
It doesn't work well in this case.